lib/auth/gitserver/gitserverv1/github.go

```diff
@@ -0,0 +1,133 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package gitserverv1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/gravitational/trace"
+
+	apidefaults "github.com/gravitational/teleport/api/defaults"
+	pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/gitserver/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth/integration/credentials"
+	"github.com/gravitational/teleport/lib/authz"
+)
+
+func (s *Service) CreateGitHubAuthRequest(ctx context.Context, in *pb.CreateGitHubAuthRequestRequest) (*types.GithubAuthRequest, error) {
+	if in.Request == nil {
+		return nil, trace.BadParameter("missing github auth request")
+	}
+	if err := types.ValidateGitHubOrganizationName(in.Organization); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	authCtx, gitServer, err := s.authAndFindServerByOrg(ctx, in.Organization)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	s.cfg.Log.DebugContext(ctx, "Creating GitHub auth request for authenticated user.",
+		"user", authCtx.User.GetName(),
+		"org", gitServer.GetGitHub().Organization,
+		"integration", gitServer.GetGitHub().Integration,
+	)
+
+	// Make a "temporary" connector spec and save it in the request.
+	uuid := uuid.NewString()
+	spec, err := s.makeGithubConnectorSpec(ctx, uuid, gitServer.GetGitHub().Organization, gitServer.GetGitHub().Integration)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	in.Request.ConnectorID = uuid
+	in.Request.ConnectorSpec = spec
```

Comment:
The ConnectorSpec is marked as **Used only in test**  https://github.com/gravitational/teleport/blob/6995108c6eeae9e61fa551feeb56577ca73c9a0c/integrations/operator/crdgen/testdata/protofiles/teleport/legacy/types/types.proto#L5197

Reply:
i did update `api/proto/teleport/legacy/types/types.proto` to indicate that `ConnectorSpec` is for both flows (basically the spec is used when not using "native" GitHub connector). I had a comment in PR description that i need to run `make -C integrations/operator crd` before merge.